gear-test/src/check.rs

```diff
@@ -422,26 +423,32 @@ pub fn check_memory(
     }
 }
 
-fn check_active_programs(
-    expected_ids: Vec<ProgramId>,
-    actual_ids: Vec<ProgramId>,
+fn check_programs_state(
+    expected_programs: &BTreeMap<ProgramId, bool>,
+    actual_programs: &BTreeMap<ProgramId, bool>,
+    exact: bool,
 ) -> Result<(), Vec<String>> {
-    let mut errors = Vec::with_capacity(expected_ids.len());
-    if expected_ids.len() != actual_ids.len() {
-        errors.push(format!(
-            "invalid amount of active programs: expected - {:?}, actual - {:?}",
-            expected_ids.len(),
-            actual_ids.len()
-        ));
-    } else {
-        let check_data = expected_ids.iter().zip(actual_ids.iter());
-        for (idx, (expected_id, actual_id)) in check_data.enumerate() {
-            if expected_id != actual_id {
+    let mut errors = Vec::new();
+
+    if exact {
+        for id in actual_programs.keys() {
```

Comment:
```suggestion
    if exact {
        // place here error pushing, please
        assert_eq!(actual_programs.len(), expected_programs.len())
        for id in actual_programs.keys() {
```

Reply:
Here you check that all active programs checked in expectations, but not that expectations contains invalid addresses. If active = [1,2] and exp = [1,2,3] that your check will be passed.

Reply:
It will fail later, in check without exact flag